addon/services/scheduler.js

```diff
@@ -54,8 +29,7 @@ const Scheduler = Service.extend({
     const token = new Token();
 
     if (queue.isActive) {
-      queue.tasks.push(callback);
-      queue.tasks.push(token);
+      queue.enqueue({token, callback});
```

Comment:
I was thinking: maybe we can have the queue manage the token (and flushing)?

That would mean this invocation would be something like `return queue.enqueue(callback)`, and `flushQueue` below would only iterate the various queues calling `queue.flush()`.  

Also, FWIW, I think depending on the number of times we do this we probably want to manage the queues internal state as an array that we traverse by twos (instead of creating intermediate objects for every callback).

Reply:
So I thought about internalizing the token and returning, but you an see that `scheduleWork` expects a token to be returned regardless of the code path. That said, the token would largely be useless if `queue.isActive === false`. 

I'm fine to keep the queue managed by twos, as I feel it's less of a code smell now than it was when exposed within the scheduler itself.